ldms/src/ldmsd/ldmsd.c

```diff
@@ -1814,23 +1815,37 @@ int main(int argc, char *argv[])
 			break;
 		case 'A':
 			/* (multiple) auth options */
-			lval = strtok(optarg, "=");
+			
+			dup_auth = strdup(optarg);
+			if (!dup_auth) {
+				printf("Not enough memory parsing options!\n");
+				free(dup_auth);
+				exit(1);
+			}
+			lval = strtok(dup_auth, "=");
 			if (!lval) {
-				printf("ERROR: Expecting -A name=value\n");
+				printf("ERROR: Expecting -A name=value. Got %s\n",
+					dup_auth);
+				free(dup_auth);
 				exit(1);
 			}
 			rval = strtok(NULL, "");
 			if (!rval) {
-				printf("ERROR: Expecting -A name=value\n");
+				printf("ERROR: Expecting -A name=value. Got %s\n",
+					dup_auth);
+				free(dup_auth);
 				exit(1);
 			}
 			if (auth_opt->count == auth_opt->size) {
-				printf("ERROR: Too many auth options\n");
+				printf("ERROR: Too many auth options %s\n",
+					dup_auth);
```

Comment:
Did you test these error paths? If everything works above, dup_auth will be the 1st word in the string that precedes the '=' sign. You should be dumping optarg in all these cases I think.

Reply:
dumping optarg would be better here. Adding the count that has been exceeded would be better also.